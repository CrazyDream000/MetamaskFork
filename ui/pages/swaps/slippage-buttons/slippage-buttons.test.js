import React from 'react';

import { renderWithProvider } from '../../../../test/jest';
import SlippageButtons from '.';

const createProps = (customProps = {}) => {
  return {
    onSelect: jest.fn(),
    maxAllowedSlippage: 15,
    currentSlippage: 3,
    ...customProps,
  };
};

describe('SlippageButtons', () => {
  it('renders the component with initial props', () => {
    const { getByText } = renderWithProvider(
      <SlippageButtons {...createProps()} />,
    );
    expect(getByText('2%')).toBeInTheDocument();
    expect(getByText('3%')).toBeInTheDocument();
    expect(getByText('custom')).toBeInTheDocument();
    expect(getByText('Advanced Options')).toBeInTheDocument();
    expect(
      document.querySelector('.slippage-buttons__header'),
    ).toMatchSnapshot();
    expect(
      document.querySelector('.slippage-buttons__button-group'),
    ).toMatchSnapshot();
  });
});
