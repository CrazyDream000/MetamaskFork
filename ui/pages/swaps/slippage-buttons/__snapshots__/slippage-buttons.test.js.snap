// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlippageButtons renders the component with initial props 1`] = `
<div
  class="slippage-buttons__header"
>
  <div
    class="slippage-buttons__header-text"
  >
    Advanced Options
  </div>
</div>
`;

exports[`SlippageButtons renders the component with initial props 2`] = `
<div
  class="button-group slippage-buttons__button-group radio-button-group"
  role="radiogroup"
>
  <button
    aria-checked="false"
    class="button-group__button radio-button"
    data-testid="button-group__button0"
    role="radio"
  >
    2%
  </button>
  <button
    aria-checked="true"
    class="button-group__button radio-button button-group__button--active radio-button--active"
    data-testid="button-group__button1"
    role="radio"
  >
    3%
  </button>
  <button
    aria-checked="false"
    class="button-group__button slippage-buttons__button-group-custom-button radio-button"
    data-testid="button-group__button2"
    role="radio"
  >
    custom
  </button>
</div>
`;
